libs/x-content/src/main/java/org/opensearch/common/xcontent/MediaType.java

```diff
@@ -62,4 +64,6 @@ public interface MediaType {
     default String typeWithSubtype() {
         return type() + "/" + subtype();
     }
+
+    XContentBuilder contentBuilder() throws IOException;
```

Comment:
@nknize interesting, it seems like some APIs were forgotten to be introduced / lost :D 

Anyway, I don't think you need `contentBuilder` but `XContent xContent()` which all media type implementation already have. Then you just need to use `XContentBuilder::builder`:

```
private static XContentBuilder createBuilder(MediaType mediaType, boolean pretty, boolean human) throws IOException {
    XContentBuilder builder = XContentBuilder.builder(mediaType.xContent());
    ...
}
```

On the other side, I believe the `MediaType`  was on purpose kept separate from `XContentType`,  may be the better refactoring would be to use `XContentType` in `Strings::toString`, not `MediaType`

Reply:
> ...I don't think you need contentBuilder but XContent xContent() which all media type implementation already have.

Nice! I did exactly this in #5902 but I like the idea of doing it up front here! Great suggestion

> I believe the MediaType was on purpose kept separate from XContentType, may be the better refactoring would be to use XContentType in Strings::toString, not MediaType

I want to decouple the XContentType implementation detail from Strings so Strings can be refactored out of server and into the library.